common/expirecontainer.h

```diff
@@ -372,3 +373,124 @@ class ObjectCache : public ObjectCacheBase {
         return borrow(key, [] { return new ValEntity(); });
     }
 };
+
+template <typename KeyType, typename NodeType>
+class ObjectCache<KeyType, intrusive_list<NodeType>> : public ObjectCacheBase {
+protected:
+    using ListType = intrusive_list<NodeType>;
+    using Base = ObjectCacheBase;
+    using KeyedItem = Base::KeyedItem<Base::Item, KeyType>;
+
+    class Item : public KeyedItem {
+    public:
+        ListType _list;
+        using KeyedItem::KeyedItem;
+        virtual Item* construct() const override {
+            auto item = new Item(this->_key);
+            item->_obj = nullptr;
+            item->_refcnt = 0;
+            item->_recycle = nullptr;
+            return item;
+        }
+        ~Item() {
+            _list.delete_all();
+        }
+    };
+
+    using ItemPtr = Item*;
+    using InterfaceKey = typename Item::InterfaceKey;
+
+public:
+    ObjectCache(uint64_t expiration) : Base(expiration, expiration / 16) {}
+    ObjectCache(uint64_t expiration, uint64_t timer_cycle)
+        : Base(expiration, timer_cycle) {}
+
+    template <typename Constructor>
+    ItemPtr ref_acquire(const InterfaceKey& key, const Constructor& ctor,
+                        uint64_t failure_cooldown = 0) {
+        auto _ctor = [&](void* item) {
+            ListType list = ctor();
```

Comment:
```((Item*)item)->_list = ctor();``` ?

Reply:
intrusive_list did not have a move constructor